src/MainWindow.vala

```diff
@@ -76,6 +76,15 @@ namespace PantheonCalculator {
             });
         }
 
+        public void undo () {
+            uint hist_length = history.length () - 1;
+            unowned List<History?> last_el = history.nth (hist_length);
```

Comment:
Apologies, I might not have been clear in my previous comments. I think it would be better to use `history.last ()` so that you can remove the `hist_length` too. I also think that `previous_entry` would be better as a variable name instead of `last_el`.